helpers/visibility.js

```diff
@@ -0,0 +1,43 @@
+const prefersReducedMotion = typeof window !== 'undefined'
+	&& window.matchMedia
+	&& window.matchMedia('(prefers-reduced-motion: reduce)').matches;
+
+export function ensureElementVisible(totalStickyOffset, scrollContainer, element) {
+	if (!element) return;
+	const rect = element.getBoundingClientRect();
+	const viewportHeight = window.innerHeight;
+
+	// Buffer for focus rings + breathing room.
+	const totalBuffer = 14;
+
+	// Determine visibility relative to sticky offset & viewport.
+	const isHiddenBySticky = (rect.top - totalBuffer) < totalStickyOffset;
+	const isOutsideViewport = (rect.bottom + totalBuffer) > viewportHeight || (rect.top - totalBuffer) < 0;
```

Comment:
Will the browser not automatically scroll to it if it's outside the viewport?

Reply:
ah yup it does; will clean this up.